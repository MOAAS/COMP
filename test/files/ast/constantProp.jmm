
class TestStack {

	public static void main(String[] args) {
		int a;
		int b;
		int c;
		int d;
		boolean x;
		boolean y;
		boolean z;
		a = 2;
		c = 5;
		b = a + 1;
		b = c + a;
		x = true;
        y = x;

        if (a < c) {

        }
        else { x = true; y = true; z = false; }

		if(a < b){
		   a = 2;
		   c = 5;
		}else{
           a = 2;
           c = 3;
		}
		b = a + c;
		a = 1;
		b = 2;
		while(a < b){
            d = 1;
            c = d;
            c = a;
            a = a + 1;
		}
		a = d;
	}
}