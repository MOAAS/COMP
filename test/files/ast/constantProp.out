Program
 ClassDeclaration
  ClassInfo
   TestStack
  MainDeclaration
   ArgName
    args
   MethodContent
    VarDeclaration
     int
     a
    VarDeclaration
     int
     b
    VarDeclaration
     int
     c
    VarDeclaration
     int
     d
    VarDeclaration
     boolean
     x
    VarDeclaration
     boolean
     y
    VarDeclaration
     boolean
     z
    Assignment
     a
     2
    Assignment
     c
     5
    Assignment
     b
     3
    Assignment
     b
     7
    Assignment
     x
     True
    Assignment
     y
     True
    If
     LessThan
      2
      b
     CodeBlock
      Assignment
       a
       2
      Assignment
       c
       5
     CodeBlock
      Assignment
       a
       2
      Assignment
       c
       3
    Assignment
     b
     Add
      2
      c
    Assignment
     a
     1
    Assignment
     b
     2
    While
     LessThan
      a
      2
     CodeBlock
      Assignment
       d
       1
      Assignment
       c
       1
      Assignment
       c
       a
      Assignment
       a
       Add
        a
        1
    Assignment
     a
     d
